### PR TITLE
Gfx940 develop

### PR DESCRIPTION
**Pull request by cgmillette**
Preparing gfx940 merges to develop branch
- Update copyright githook
- Add gfx940:xnack-/+ targets
- Add non-native types f8, bf8 and xf32 
- Add mfma support for f8, bf8 and xf32 types

### CODE REVIEW

File: library/include/rocwmma/internal/config.hpp [hunk 1]
```diff
@@ -34,20 +34,25 @@ namespace rocwmma
...
 #define ROCWMMA_ARCH_GFX90A __gfx90a__
+#elif defined(__gfx940__)
+#define ROCWMMA_ARCH_GFX940 __gfx940__
 #elif defined(__gfx1100__)
 #define ROCWMMA_ARCH_GFX1100 __gfx1100__
 #elif defined(__gfx1101__)
 #define ROCWMMA_ARCH_GFX1101 __gfx1101__
 #elif defined(__gfx1102__)
 #define ROCWMMA_ARCH_GFX1102 __gfx1102__
+#elif defined(__gfx940__)
```
Comment (cgmillette) [inline library/include/rocwmma/internal/config.hpp:54]: This is duplicated and can be removed

File: library/include/rocwmma/internal/config.hpp [hunk 2]
```diff
@@ -67,6 +75,9 @@ namespace rocwmma
 #if !defined(ROCWMMA_ARCH_GFX1102)
 #define ROCWMMA_ARCH_GFX1102 0
 #endif
+#if !defined(ROCWMMA_ARCH_GFX940)
```
Comment (cgmillette) [inline library/include/rocwmma/internal/config.hpp:78]: Duplicated and can be removed

**Review by mkarunan — APPROVED** [review]
rocm-smi errors due to uint64_t, should we default to ::uint64_t;

**Comment by cgmillette** [comment]
> rocm-smi errors due to uint64_t, should we default to ::uint64_t;

Yep, covered in the latest commit

**Review by dlangbe — APPROVED** [review]

**Pull request merged**